tools/deviceGen/input/device_features.json

```diff
@@ -0,0 +1,2666 @@
+            "comment": "[REQUIRE] Roadmap 2022 requires support for these, device support is ubiquitous"
+        },
+        {
+            "groupComment": [
+                "[DO NOT EXPOSE] these 3 don't make a difference, just shortcut from Querying support from PhysicalDevice"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "textureCompressionETC2"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "textureCompressionASTC_LDR"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "textureCompressionBC"
+        },
+        {
+            "type": "bool",
+            "name": "occlusionQueryPrecise",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2022 and good device support"
+        },
+        {
+            "type": "bool",
+            "name": "pipelineStatisticsQuery",
+            "value": false
+        },
+        {
+            "groupComment": [
+                "[EXPOSE AS LIMIT] Enabled by Default, Moved to Limits"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "vertexPipelineStoresAndAtomics"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "fragmentStoresAndAtomics"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderTessellationAndGeometryPointSize"
+        },
+        {
+            "type": "bool",
+            "name": "shaderImageGatherExtended",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2024 good device support"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageExtendedFormats",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2024 good device support"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderStorageImageMultisample",
+            "comment": "[EXPOSE AS LIMIT] Cannot be always enabled cause Intel ARC is handicapped"
+        },
+        {
+            "groupComment": [
+                "TODO: format feature reporting unimplemented yet for both of the below! (should we move to usage reporting?)",
+                "[EXPOSE AS LIMIT] always enable, shouldn't cause overhead by just being enabled"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderStorageImageReadWithoutFormat"
+        },
+        {
+            "type": "bool",
+            "name": "shaderStorageImageWriteWithoutFormat",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] good device support"
+        },
```

Comment:
and https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#r1641936421